### PR TITLE
Export orders and order lines

### PR DESCRIPTION
**Pull request by nvthanh11**
Add option to export orders and order lines

### CODE REVIEW

File: shuup/admin/modules/orders/__init__.py [hunk 1]
```diff
@@ -207,3 +207,27 @@ def get_menu_entries(self, request):
 
     def get_model_url(self, object, kind, shop=None):
         return derive_model_url(OrderStatus, "shuup_admin:order_status", object, kind)
+
+
+class ShipmentModule(AdminModule):
```
Comment (chessbr) [inline shuup/admin/modules/orders/__init__.py:212]: why this is here?

File: shuup/admin/modules/orders/views/list.py [hunk 2]
```diff
@@ -91,3 +91,161 @@ def get_object_abstract(self, instance, item):
...
+    ChoicesFilter, Column, DateRangeFilter, MultiFieldTextFilter, RangeFilter,
+    TextFilter
+)
+from shuup.admin.utils.views import PicotableListView
+from shuup.core.models import Order, OrderStatus, PaymentStatus, ShippingStatus, Shipment, ShipmentStatus, ShipmentProduct
+from shuup.utils.django_compat import reverse
+from shuup.utils.i18n import format_money, get_locally_formatted_datetime
+
+
+class OrderListView(PicotableListView):
```
Comment (chessbr) [inline shuup/admin/modules/orders/views/list.py:121]: why this is here?

File: shuup/default_reports/reports/orders.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                "order_line_sku": line.sku,
+                "order_line_text": line.text,
+                "order_line_quantity": line.quantity,
+                "taxless_unit_price": line.taxless_base_unit_price,
+                "taxful_unit_price": line.taxful_base_unit_price,
+                "taxful_price": line.taxful_price,
+                "type": line.type.name.capitalize(),
+                "created_on": get_locally_formatted_datetime(line.created_on),
+            })
+        return self.get_return_data(data, has_totals=False)
```
Comment (chessbr) [inline shuup/default_reports/reports/orders.py:82]: add missing blank line at the end of the file

File: shuup/reports/settings.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+
+# report
+DEFAULT_REPORTS_ITEM_LIMIT = 2000
```
Comment (chessbr) [inline shuup/reports/settings.py:10]: Add a good documentation about what this setting is for and add missing blank line at the end of the file

File: shuup/default_reports/mixins.py [hunk 5]
```diff
@@ -26,5 +26,24 @@ def get_objects(self):
...
+        types = self.options.get("order_line_type")
+        order_status = self.options.get("order_status")
+        filters = Q()
+        if supplier:
+            filters &= Q(supplier__in=supplier)
+        if types:
+            filters &= Q(type__in=types)
+        if order_status:
+            filters &= Q(order__status__in=order_status)
+        return queryset.filter(filters).order_by("created_on")
```
Comment (chessbr) [inline shuup/default_reports/mixins.py:49]: add missing blank line at the end of the file

File: shuup/default_reports/reports/orders.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        data = []
+        orders = self.get_objects(paid=False)
+
+        for order in orders:
+            data.append({
+                "order_num": order.identifier,
+                "order_date": get_locally_formatted_datetime(order.order_date),
+                "status": order.status,
+                "order_line_quantity": order.lines.filter(type=OrderLineType.PRODUCT).count(),
+                "order_total_amount": order.taxful_total_price,
```
Comment (chessbr) [inline shuup/default_reports/reports/orders.py:43]: use `format_money` function here
```py
from shuup.utils.i18n import format_money

...

"order_total_amount": format_money(order.taxful_total_price),
```

File: shuup/default_reports/reports/orders.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+    def get_data(self):
+        data = []
+        order_lines = self.get_objects()[:self.queryset_row_limit]
+        for line in order_lines:
+            data.append({
+                "order_line_sku": line.sku,
+                "order_line_text": line.text,
+                "order_line_quantity": line.quantity,
+                "taxless_unit_price": line.taxless_base_unit_price,
```
Comment (chessbr) [inline shuup/default_reports/reports/orders.py:76]: use `format_money` function on the prices below

**Review by chessbr — CHANGES_REQUESTED** [review]

File: shuup/admin/modules/orders/__init__.py (hunk 1 above)
Comment (nvthanh11) [reply shuup/admin/modules/orders/__init__.py:212]: This is the implementation for Shippents entry button, and it's a child of Order  Admin module so I put it here. Shipment doesn't have its own module

Comment (nvthanh11) [reply shuup/admin/modules/orders/__init__.py:212]: sorry. I pushed code to a wrong branch

File: shuup/admin/modules/orders/views/list.py (hunk 2 above)
Comment (nvthanh11) [reply shuup/admin/modules/orders/views/list.py:121]: sorry. I pushed code to a wrong branch

**Comment by nvthanh11** [comment]
I'll create new PR for this

**Pull request closed without merging**